unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/ParcelScene.cs

```diff
@@ -1174,94 +1114,51 @@ private DecentralandEntity GetEntityForUpdate(string entityId)
             return decentralandEntity;
         }
 
-        private void OnDisposableReady(BaseDisposable disposable)
+        private void DisposeAllSceneComponents()
         {
-            if (isReleased)
-                return;
-
-            disposableNotReady.Remove(disposable.id);
-
-            if (VERBOSE)
-            {
-                Debug.Log($"{sceneData.basePosition} Disposable objects left... {disposableNotReady.Count}");
-            }
-
-            if (disposableNotReady.Count == 0)
+            List<string> allDisposableComponents = disposableComponents.Select(x => x.Key).ToList();
+            foreach (string id in allDisposableComponents)
             {
-                SetSceneReady();
+                Environment.i.parcelScenesCleaner.MarkDisposableComponentForCleanup(this, id);
             }
```

Comment:
Why not use `disposableComponents.Keys` here instead of doing a `Select` + `ToList`?

Reply:
Since we are marking the disposables for cleanup (not destroying them immediately) I also think we should just traverse the `disposableComponents` existent dictionary keys instead of creating a strings list here.

Reply:
The request is valid but we should address it in another PR. This PR is just to move the existing code to another class.